williamrijksen/com.williamrijksen.onesignal
# Introduce permission listeners

## Context
Introduces the permission changes listeners to the appcelerator module
 - [addPermissionObserver iOS](https://documentation.onesignal.com/docs/ios-native-sdk#section--addpermissionobserver-)
 - [addPermissionObserver android](https://documentation.onesignal.com/docs/android-native-sdk#section--addpermissionobserver-)

## How to test
 - build the module
 - add the following code to your test device/application
```js
        var onesignal = require('com.williamrijksen.onesignal');
        
        onesignal.addEventListener('permissionChanged', function (evt) {
            console.log(JSON.stringify(evt));
        });
```
- Verify the outcome is like this:
```js
// evt iOS
{"to":{"status":2,"hasPrompted":true},"from":{"status":1,"hasPrompted":true}}
// evt android
{"to":{"enabled":true},"from":{"enabled":false}}
```

## Notes
 Blocked by #62
